apis/k8ssandra/v1alpha1/k8ssandracluster_types.go

```diff
@@ -457,26 +461,84 @@ type EmbeddedObjectMeta struct {
 	Name string `json:"name"`
 
 	// +optional
-	Labels map[string]string `json:"labels,omitempty"`
+	Metadata *CassandraDatacenterMeta `json:"dcMetadata,omitempty"`
```

Comment:
It would be nice to not have both `metadata` and `dcMetadata` here. With a bit of refactoring, I think this is achievable:

```go
type EmbeddedObjectMeta struct {
	Namespace string `json:"namespace,omitempty"`
	Name string `json:"name"`
	CassandraDatacenterMeta `json:",inline"`
}
```

The beauty of it is that you can then expose only `CassandraDatacenterMeta` at cluster-level, and the whole `EmbeddedObjectMeta` at DC level. This solves the issue of not exposing `name` and `namespace` at cluster level:

```go
type CassandraClusterTemplate struct {
	DatacenterOptions `json:",inline"`
	Meta CassandraDatacenterMeta `json:"metadata,omitempty"`
	// ...
}

type CassandraDatacenterTemplate struct {
	DatacenterOptions `json:",inline"`
	Meta EmbeddedObjectMeta `json:"metadata,omitempty"`
        // ...
}
```

Merging the 2 fields can be done this way:

```go
	cassandraDatacenterMeta := goalesceutils.MergeCRs(clusterTemplate.Meta, dcTemplate.Meta.CassandraDatacenterMeta)
	dcConfig.Meta.CassandraDatacenterMeta = cassandraDatacenterMeta
```